### PR TITLE
Add windows command and logic to choose command based on platform

### PR DESCRIPTION
**Pull request by tew9ey**
Hello @NickLiffen,
I added the windows command and they were able to work correctly.
- In addition to the standard commands, I added more commands to allow to purge the temGitLocation after the commands are done running on each repo in repo.json to allow recursive execution on more than one repos in repos.json.

- Created the logic, and it's working perfectly on my windows but haven't implemented the error handling properly for the case the machine is neither windows nor mac, If you can change that line of code to give a friendly message that'd be great.

Feel free to reach out if you have any question

### CODE REVIEW

**Review by tew9ey — COMMENTED** [review]
@NickLiffen I like the approach of using execFile child process for the windows command.

**Comment by NickLiffen** [comment]
> @NickLiffen I like the approach of using execFile child process for the windows command.

Thanks 👍 Need to test it works 💯 but I think the foundation should do it 💯

**Comment by tew9ey** [comment]
> > @NickLiffen I like the approach of using execFile child process for the windows command.
> 
> Thanks 👍 Need to test it works 💯 but I think the foundation should do it 💯

Will test it now and let you know if there's any issue.

**Comment by NickLiffen** [comment]
> Will test it now and let you know if there's any issue.

That would be amazing 💯 today is my first day back so im just catching up with things, if you could test that would be AMAZING 👍 thank you :)

**Comment by tew9ey** [comment]
> > Will test it now and let you know if there's any issue.
> 
> That would be amazing 💯 today is my first day back so im just catching up with things, if you could test that would be AMAZING 👍 thank you :)

I tested it on windows, but it was breaking due to the new changes in the commit.ts file "execfile", It'll require the windows commands to be split into commands and arguments like you did with mac commands. I tested it with mkdir command splitter it to mkdir and args: '-p', ${tempdir' and then that specific command got rid of Spawn error I was encountering at first.

**Comment by tew9ey** [comment]
errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn git clone https://github.com/EY-POC/dev-secOps.git',
  path: 'git clone https://github.com/EY-POC/dev-secOps.git',
  spawnargs: [],
  cmd: 'git clone https://github.com/EY-POC/dev-secOps.git',
  stdout: '',
  stderr: ''
}

That's the error it was throwing, I can work on it on my convenient time.

**Comment by NickLiffen** [comment]
@tew9ey just checked and this works on a Mac 👍 The changes I made seem to work well 👍 

Is there any chance you could update your Windows commands section and then test it 👍 Once that works we can merge this in :) Thanks for all your support and help :)

**Comment by NickLiffen** [comment]
@tew9ey I made some global changes based on what you proposed. I loved some of the idea 👍 You should just be able to update the commands, test it, let me know and we can get this merged 👍 

Once again, thanks for working on this with me :)

**Comment by NickLiffen** [comment]
@tew9ey 👋 

If you have time could you test the changes I just made 👋 They _may_ work (they may not) there are some funky `\/` in there that I wouldn't know for sure would work without actually running them. It does look like it solved the security vulnerability though.

**Comment by tew9ey** [comment]
@NickLiffen in windows using "\\" as the split delimiter and using Documents as the directory fixes the spawn error, can you review the changes I just pushed? they're working on windows, I've tested them on mine and other team members.

**Comment by NickLiffen** [comment]
AMAZING 👯 This is really nice work 💯 

Thanks so much for helping me with this. Sending you lots of ✨ ✨ ✨ 

Really nice :)

**Review by NickLiffen — APPROVED** [review]
lgtm

**Comment by tew9ey** [comment]
> AMAZING 👯 This is really nice work 💯
> 
> Thanks so much for helping me with this. Sending you lots of ✨ ✨ ✨
> 
> Really nice :)

Thank you so much @NickLiffen, you've put so much effort into this. letting you know that I appreciate that as an end user.

File: src/utils/commitFile.ts [hunk 1]
```diff
@@ -3,16 +3,26 @@
...
 import { response, commands } from "../../types/common";
 
 const exec = util.promisify(ImportedExec);
 
+const platform = os.platform();
+
+const isWindows = platform === "win32";
+if (platform !== "win32" && platform !== "darwin") {
+  error("You can only use either windows or mac machine!");
+  throw Error;
```
[the remark this answers is not in the document]
Comment (tew9ey) [reply src/utils/commitFile.ts]: Love it, it's more clearer

**Comment by NickLiffen** [comment]
> Thank you so much @NickLiffen, you've put so much effort into this. letting you know that I appreciate that as an end user.

No problem 👍 What I am hear for 💯 It's nice getting contributions from people who use the tool so we can make it better :) Have a great weekend and I'm sure we will chat soon.

**Comment by tew9ey** [comment]
> > Thank you so much @NickLiffen, you've put so much effort into this. letting you know that I appreciate that as an end user.
> 
> No problem 👍 What I am hear for 💯 It's nice getting contributions from people who use the tool so we can make it better :) Have a great weekend and I'm sure we will chat soon.

you too, we will definitely do.

**Pull request merged**